Updates to allow manual build of conda packages on M1

cticenhour opened the pull request:
This PR provides the settings and configuration necessary for local builds of `moose-mpich`, `moose-libmesh-vtk`, `moose-petsc`, `moose-libmesh`, and `moose-tools` on M1 Apple Silicon. This *does not* mean they are ready for regular use, as we cannot yet test them across the full range of CIVET configurations normal for such macOS packages.

All MOOSE tests for these local packages pass with `opt` in both framework and unit tests as of the current next branch. Four geochemistry tests remain broken (see below for a results summary), and extended testing on heavy tests (in general) still needs to be performed.  

```
Modules:
geochemistry/test:equilibrium_models.ph_constraint .......................... FAILED (EXPECTED OUTPUT MISSING)
geochemistry/test:equilibrium_models.seawater_no_precip ..................... FAILED (EXPECTED OUTPUT MISSING)
geochemistry/test:time_dependent_reactions.flushing ........................................... FAILED (CRASH)
geochemistry/test:time_dependent_reactions.mixing_seawater_step2 ............................ FAILED (CSVDIFF)
--------------------------------------------------------------------------------------------------------------
Ran 4526 tests in 652.0 seconds. Average test time 0.9 seconds, maximum test time 29.7 seconds.
4522 passed, 366 skipped, 0 pending, 4 FAILED
```

Refs #18954

cticenhour committed:
Changes to partially build PETSc and tune PETSc/MPICH on M1 (#18954)

- Builds up to PETSc linking with SLEPc, which then fails.
- Adds Apple CPU tuning to MPICH build:
  - Default for Apple Silicon in llvm-12.0.1 was apple-a12.
  - Using this as `mcpu` did not work for FORTRAN, so sticking with the nearest compatible (armv8.3-a) march for those tuning variables.
- Add Apple CPU tuning to PETSC build

Commit:
PETSc now working on ARM, push to test on Linux, Darwin (#18954)

Commit:
Remove shared config, doing it in config_petsc.sh

Commit:
Adding a possibly needed build dependency

adding m4 to libMesh's build dependencies.

Commit:
Build SLEPc once more

Who knows why this was failing before. Working now.

cticenhour committed:
Update conda package builds (#18954)

cticenhour committed:
Update M1 developer conda stack build instructions (#18954)

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/937501/) on db8c640 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/20103/site)

This comment will be updated on new commits.

cticenhour commented on modules/doc/content/getting_started/installation/m1_monterey.md:
```diff
@@ -2,15 +2,15 @@
-Then the [MOOSE installation instructions for conda](conda.md) can be used to perform
-the installation.
+!alert! note title=Mambaforge3
+This guide assumes that you have mambaforge3 installed. The [MOOSE installation
+instructions for conda](conda.md) can be used to perform the installation. Note
```
```suggestion
This guide assumes that you have mambaforge3 installed. The [MOOSE installation instructions for conda](conda.md) 
can be used to perform the installation. Note
```

cticenhour committed:
Reorganize logic and remove unrecognized flag (#18954)

An alternate form, headerpad_max_install_names, is already available by 
default judging from the conda build output.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/938031/) on 5b30c01 wanted to post the following:


## Framework coverage

Coverage did not change


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/20103/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/20103/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/20103/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/20103/coverage/contact)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/20103/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/20103/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/20103/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/20103/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/20103/coverage/geochemistry)
- [`heat_conduction`](https://mooseframework.inl.gov/docs/PRs/20103/coverage/heat_conduction)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/20103/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/20103/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/20103/coverage/navier_stokes)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/20103/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/20103/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/20103/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/20103/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/20103/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/20103/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/20103/coverage/richards)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/20103/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/20103/coverage/tensor_mechanics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/20103/coverage/xfem)

</details>



This comment will be updated on new commits.

loganharbour commented on scripts/configure_petsc.sh:
```diff
@@ -44,7 +50,7 @@ function configure_petsc()
       --with-sowing=0 \
       --with-64-bit-indices \
-      $*
+      "$@"
```
Back to this?

cticenhour replied:
Yeah, we keep running into needing it, so it probably needs to remain....at least the quotes. We might be able to get by with `

milljm reviewed (approved):
👍
